docker/scripts/setup_env

```diff
@@ -44,34 +50,40 @@ DATABASE_ENCODING=${HUGINN_DATABASE_ENCODING:-${DATABASE_ENCODING}}
 DATABASE_PASSWORD=${HUGINN_DATABASE_PASSWORD:-${DATABASE_PASSWORD:-password}}
 DATABASE_NAME=${HUGINN_DATABASE_NAME:-${DATABASE_NAME:-huginn_production}}
 RAILS_ENV=${HUGINN_RAILS_ENV:-${RAILS_ENV:-production}}
+RAILS_LOG_TO_STDOUT=${RAILS_LOG_TO_STDOUT:-true}
+RAILS_SERVE_STATIC_FILES=${RAILS_SERVE_STATIC_FILES:-true}
+IP="${IP:-0.0.0.0}"
+PORT=${HUGINN_PORT:-${PORT:-3000}}
+PORT="$(echo $PORT|awk -F ':' '{print $NF}')"
+DOMAIN=${DOMAIN:-$IP:$PORT}
 
 # Add ENV variables to .env.example which are not present in it but usable
-cat >> /app/.env.example <<EOF
+cp -a /app/.env.example /tmp/
+cat >> /tmp/.env.example <<EOF
 ASSET_HOST=
 DEFAULT_SCENARIO_FILE=
+IP=
+RAILS_ENV=
+RAILS_LOG_TO_STDOUT=
 RAILS_SERVE_STATIC_FILES=
 SEED_EMAIL=
 SEED_PASSWORD=
 SEED_USERNAME=
 SMTP_OPENSSL_CA_FILE=
 SMTP_OPENSSL_CA_PATH=
 SMTP_OPENSSL_VERIFY_MODE=
+START_MYSQL=
 EOF
 
 # Default to the environment variable values set in .env.example
 IFS="="
-sed -n -r -e 's/^#?([A-Za-z0-9_]+ *=.*)/\1/p' /app/.env.example | \
+sed -n -r -e 's/^ *#?([A-Za-z0-9_]+=.*)/\1/p' /tmp/.env.example | \
```

Comment:
Is this change needed? afaik we don't have lines starting with a space in `.env.example`.

Reply:
Well, in someone...'s local version, they may decide to add lines with spaces at the beginning (e.g., '[space][space]ABC=abc'). This is correct; you can source it in bash for example. On the other hand, 'ABC[space]=abc' is incorrect. This change basically reflects the correct parsing behavior.

Reply:
Hm the dotenv gem we use to load the `.env` file in non docker setup does not allow leading spaces in the version we are using currently. I'd like the two parsing methods to behave the same, so i'd say either we keep disallowing the leading space or bump the `dotenv` version.

> On the other hand, 'ABC[space]=abc' is incorrect

Makes sense.

Reply:
I forgot to mention (because the comment is hidden now) that the regex was modified to not allow leading white space.